tests/raster/test_netcdf.py

```diff
@@ -0,0 +1,548 @@
+            np.testing.assert_equal(ts1.data[data_variables[0]], data[:, :10, :10])
+            np.testing.assert_equal(ts2.data[data_variables[0]], data[:, 10:15, 12:17])
+
+        data = data * 2. + 3.
+        with NetCdfReader.from_filepaths(filepaths) as nc_reader:
+            ts1 = nc_reader.select_px_window(0, 0, width=10, height=10, inplace=False)
+            ts1.read(data_variables=data_variables, auto_decode=True)
+            ts2 = nc_reader.select_px_window(10, 12, width=5, height=5, inplace=False)
+            ts2.read(data_variables=data_variables, auto_decode=True)
+            np.testing.assert_equal(ts1.data[data_variables[0]], data[:, :10, :10])
+            np.testing.assert_equal(ts2.data[data_variables[0]], data[:, 10:15, 12:17])
+
+    def test_write_selections(self):
+        """ Tests writing mosaic after some select operations have been applied. """
+        num_files = 10
+        xsize = 60
+        ysize = 50
+        layer_ids = [0, 5, 9]
+        data_variable = 'mosaic'
+
+        ds_tile = Tile(ysize, xsize, SpatialRef(4326), name='0')
+        ds_mosaic = MosaicGeometry.from_tile_list([ds_tile])
+
+        dims = ['time', 'y', 'x']
+        dates = pd.date_range('2000-01-01', periods=num_files)
+        coords = {'time': pd.date_range('2000-01-01', periods=num_files),
+                  'y': ds_tile.y_coords,
+                  'x': ds_tile.x_coords}
+        data = np.random.randn(num_files, ysize, xsize)
+        ds = xr.Dataset({'mosaic': (dims, data)}, coords=coords)
+        dst_filepath = os.path.join(self.path, "test.nc")
+        layers = dates[layer_ids]
+        with NetCdfWriter.from_data(ds, dst_filepath, mosaic=ds_mosaic, stack_dimension='time') as nc_writer:
+            nc_writer.select_layers(layers, inplace=True)
+            ul_data = nc_writer.select_px_window(0, 0, height=25, width=30)
+            ur_data = nc_writer.select_px_window(0, 30, height=25, width=30)
+            ll_data = nc_writer.select_px_window(25, 0, height=25, width=30)
+            lr_data = nc_writer.select_px_window(25, 30, height=25, width=30)
+            nc_writer.write(ul_data.data)
+            nc_writer.write(ur_data.data)
+            nc_writer.write(ll_data.data)
+            nc_writer.write(lr_data.data)
+            filepaths = list(set(nc_writer.file_register['filepath']))
+
+        with NetCdfReader.from_filepaths(filepaths) as nc_reader:
+            ts1 = nc_reader.select_px_window(0, 0, width=10, height=10, inplace=False)
+            ts1.read(data_variables=data_variable)
+            ts2 = nc_reader.select_px_window(45, 55, width=5, height=5, inplace=False)
+            ts2.read(data_variables=data_variable)
+            np.testing.assert_equal(ts1.data[data_variable], data[layer_ids, :10, :10])
+            np.testing.assert_equal(ts2.data[data_variable], data[layer_ids, 45:, 55:])
+
+    def tearDown(self):
+        """
+        Remove test file.
+        """
+        shutil.rmtree(self.path)
```

Comment:
same as geotiff, and probably also parameterizeable to reduce code duplication

Reply:
Done!